packages/ui/src/utils/ClipboardManager.ts

```diff
@@ -0,0 +1,45 @@
+import { IClipboardCopyObject } from '../components/Visualization/Custom/hooks/copy-step.hook';
+import { updateIds } from './update-ids';
+
+export class ClipboardManager {
+  private static clipboardContent: string | null = null;
+
+  /**
+   * Serialize an object into a JSON string and store it in the clipboardContent.
+   * @param object The object to serialize.
+   */
+  static copy(object: IClipboardCopyObject): void {
+    try {
+      this.clipboardContent = JSON.stringify(object);
+      navigator.clipboard.writeText(this.clipboardContent);
+    } catch (err) {
+      console.error('Failed to copy to clipboard:', err);
+      throw new Error('Clipboard copy failed');
+    }
+  }
+
+  /**
+   * Deserialize the clipboardContent into an object.
+   * @returns The deserialized object.
+   */
+  static paste(): IClipboardCopyObject | null {
+    if (!this.clipboardContent) {
+      return null;
+    }
+
+    try {
```

Comment:
We need to perform validations here. Given a defined structure, we should check that the object we want to paste has that structure; otherwise, we might be pasting something unintended.

Reply:
We definitely need a defined structure for the copied content. It could look something like this:
```
{
  "kaotoContent": true, // optional
  "entityType": "route",
  "copiedComponentName": "choice",
  "DSL": <Copied Node Definition>
}
```
However, there's a catch - during paste, we don't actually read the system clipboard directly. Instead, we use `ClipboardManager.clipboardContent`, which kind of eliminates the need for runtime validation, in my opinion.

What’s your take on this approach?

Reply:
I think I'm missing something. How is the runtime validation eliminated?